api/api.go

```diff
@@ -1,51 +1,102 @@
 package api
 
 import (
-	"bytes"
-	"io/ioutil"
-
-	"golang.org/x/net/html/charset"
-	"golang.org/x/text/transform"
-)
-
-const (
-	mimeType = "text/plain"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
 )
 
 var (
-	utf8BOM = []byte{0xef, 0xbb, 0xbf}
+	// UserAgent lets the API know where the call is being made from.
+	// It's overridden from the root command so that we can set the version.
+	UserAgent = "github.com/exercism/cli"
+
+	// HTTPClient is the client used to make HTTP calls in the cli package.
+	HTTPClient = &http.Client{Timeout: 60 * time.Second}
 )
 
-func readFileAsUTF8String(filename string) (*string, error) {
-	b, err := ioutil.ReadFile(filename)
+// Client is an http client that is configured for Exercism.
+type Client struct {
+	APIBaseURL  string
+	ContentType string
+	Token       string
+	*http.Client
+}
+
+// NewClient returns an Exercism API client.
+func NewClient(token, baseURL string) (*Client, error) {
+	return &Client{
+		APIBaseURL: baseURL,
+		Token:      token,
+		Client:     HTTPClient,
+	}, nil
+}
+
+// NewRequest returns an http.Request with information for the Exercism API.
+func (c *Client) NewRequest(method, url string, body io.Reader) (*http.Request, error) {
+	if c.Client == nil {
+		c.Client = HTTPClient
+	}
+
+	req, err := http.NewRequest(method, url, body)
 	if err != nil {
 		return nil, err
 	}
 
-	encoding, _, certain := charset.DetermineEncoding(b, mimeType)
-	if !certain {
-		// We don't want to use an uncertain encoding.
-		// In particular, doing that may mangle UTF-8 files
-		// that have only ASCII in their first 1024 bytes.
-		// See https://github.com/exercism/cli/issues/309.
-		// So if we're unsure, use UTF-8 (no transformation).
-		s := string(b)
-		return &s, nil
```

Comment:
Are we losing something here with the removal of the uncertain encoding check? Or was it not being used?

Reply:
I didn’t see it being used anywhere so I removed it. It seems like an artifact from CLI 2, but I have yet to confirm.

Reply:
Ooooh, this makes sense. We used to serialize the files into JSON. Now we're just sending the raw files.